crypto/params.c

```diff
@@ -791,42 +796,51 @@ int OSSL_PARAM_set_int64(OSSL_PARAM *p, int64_t val)
             err_out_of_range;
             return 0;
         case sizeof(int64_t):
+            p->return_size = sizeof(int64_t);
             *(int64_t *)p->data = val;
             return 1;
         }
 #endif
         return general_set_int(p, &val, sizeof(val));
     } else if (p->data_type == OSSL_PARAM_UNSIGNED_INTEGER && val >= 0) {
 #ifndef OPENSSL_SMALL_FOOTPRINT
-        p->return_size = sizeof(uint64_t); /* Expected size */
-        if (p->data == NULL)
+        if (p->data == NULL) {
+            p->return_size = sizeof(uint64_t); /* Expected size */
             return 1;
+        }
         switch (p->data_size) {
         case sizeof(uint32_t):
-            if (val <= UINT32_MAX) {
+            if (val >= 0 && val <= UINT32_MAX) {
```

Comment:
The `val >= 0` check is redundant here which is why it wasn't included originally.
It's done above (line 805).

Reply:
That branch condition went unnoticed, thanks.